docs/proposals/diamond-design.md

```diff
@@ -0,0 +1,562 @@
+   This command fails when when the split has not been already created.
+6. Incremental uploads for a bundle and diamonds work the same way. See [below](#diamonds-vs-incremental-uploads)
+
+
+> **Trade-off**: the possibility given to the user to opt-in for a specific `{split-id}` reflects the desire
+> to avoid "technical conflicts" whenever needed.
+>
+> In all situations not requiring this kind of handling, not caring about this id is the recommended way.
+
+### Commit as a bundle
+
+Eventually, all file lists combine into a new bundle with the commit message and possibly a label.
+
+1. The initiator of the diamond **needs to coordinate with the different clients to decide a commit** and create a new bundle.
+2. Any `split add` operation still in progress at commit time will be ignored in the resulting bundle.
+3. Command:
+   ```
+   datamon diamond commit --repo {repo} --diamond {id} --message {message} [--label {label}]
+   ```
+4. Optionally a commit can be set to fail on conflict.
+   ```
+   datamon diamond commit --repo {repo} --diamond {diamond-id} \
+                          --message {message} --label {label} --no-conflicts
+   ```
+   In that case, the conflicting splits are indicated in the error message.
+5. **Nice to have**. Alternatively, the whole diamond operation may be canceled (see [§ Cancellation](#handling-failures-retries-and-cancellations)).
+   This allows for some housekeeping of the `vmetadata` bucket (see below)
+
+
+### Metadata housekeeping
+
+**Nice to have**. A background job periodically cleans up metadata for all terminated diamonds and their splits.
+
+1. Regular users and service accounts are not authorized to delete objects in buckets.
+2. After a commit or cancel has been carried out, we may safely remove all diamond-related metata.
+3. We don't want metadata buckets to grow and hold forever what is essentially temporary.
+4. A privileged job is in responsible for cleaning, for all repositories in a given context, all references to terminated
+   diamonds and their splits in the vmetadata bucket.
+
+> **NOTE**: while not critical, this task guards our `vmetadata` bucket against undesirable pollution from tracks of
+> failed CI jobs, etc. We could run this on every week's end for instance.
+
+
+## Example
+
+```bash
+repo="test-repo"
+diamond=$(datamon diamond initialize --repo "${repo}")
+
+datamon diamond add --repo "${repo}" --diamond "${diamond-id}" --path /datasource/part1 &
+datamon diamond add --repo "${repo}" --diamond "${diamond-id}" --path /datasource/part2 &
+datamon diamond add --repo "${repo}" --diamond "${diamond-id}" --path /datasource/part3 &
+
+datamon diamond split list --repo "${repo}" --diamond "${diamond-id}"
+
+datamon diamond commit --repo "${repo}" --diamond "${diamond-id}" --message "Diamond constructed"
+```
+
+
+## Detailed design
+
+### Diamonds vs incremental uploads
+
+Diamonds and incremental uploads are very much alike, the use-cases come with some nuances.
+
+Use-cases:
```

Comment:
i want to note here for thoroughness that adding some additional verbiage to the cli for the incremental upload from a single client use-case in particular rather than expecting desktop users to be familiar with all concepts exposed by the diamond cli additions.